File: .skills/openclaw-skills/skills/chair4ce/swarm/lib/providers/groq.js
/**
 * Groq Provider
 * FREE tier available! Very fast inference.
 * Llama 3.1 70B, Mixtral, etc.
 */

const { BaseLLMProvider } = require('./base');

class GroqProvider extends BaseLLMProvider {
  constructor(config = {}) {
    super(config);
    this.name = 'groq';
    this.model = config.model || 'llama-3.1-70b-versatile';
    this.endpoint = 'https://api.groq.com/openai/v1/chat/completions';
    
    // Groq pricing (very cheap, has free tier)
    this.pricing = {
      'llama-3.1-70b-versatile': { input: 0.59, output: 0.79 },
      'llama-3.1-8b-instant': { input: 0.05, output: 0.08 },
      'mixtral-8x7b-32768': { input: 0.24, output: 0.24 },
      'gemma2-9b-it': { input: 0.20, output: 0.20 },
    };
  }

  async complete(prompt, options = {}) {
    const response = await fetch(this.endpoint, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        'Authorization': `Bearer ${this.apiKey}`,
      },
      body: JSON.stringify({
        model: this.model,
        messages: [{ role: 'user', content: prompt }],
        max_tokens: options.maxTokens || this.maxTokens,
        temperature: options.temperature || this.temperature,
      }),
    });

    if (!response.ok) {
      const error = await response.text();
      throw new Error(`Groq API error: ${response.status} - ${error}`);
    }

    const data = await response.json();
    
    // Track usage
    const usage = data.usage || {};
    this.trackTokens(usage.prompt_tokens || 0, usage.completion_tokens || 0);
    
    return data.choices[0].message.content;
  }

  getEstimatedCost() {
    const modelPricing = this.pricing[this.model] || this.pricing['llama-3.1-70b-versatile'];
    const inputCost = (this.tokenCount.input / 1_000_000) * modelPricing.input;
    const outputCost = (this.tokenCount.output / 1_000_000) * modelPricing.output;
    return {
      inputTokens: this.tokenCount.input,
      outputTokens: this.tokenCount.output,
      inputCost: inputCost.toFixed(6),
      outputCost: outputCost.toFixed(6),
      totalCost: (inputCost + outputCost).toFixed(6),
      currency: 'USD',
      note: 'Groq has a free tier with rate limits',
    };
  }
}

module.exports = { GroqProvider };

File: .skills/openclaw-skills/skills/samirjtv-ctrl/groq-2/scripts/groq.py
# groq.py

import requests
import os

def groq_completion(prompt):
    api_key = os.environ.get("GROQ_API_KEY")
    if not api_key:
        return "Error: GROQ_API_KEY not set."

    invoke_url = "https://api.groq.com/openai/v1/chat/completions"
    headers = {
        "Content-Type": "application/json",
        "Authorization": f"Bearer {api_key}"
    }
    payload = {
        "model": "openai/gpt-oss-120b",
        "messages": [{"role": "user", "content": prompt}],
        "max_tokens": 512,
        "temperature": 1.00,
        "top_p": 1.00,
        "frequency_penalty": 0.00,
        "presence_penalty": 0.00,
        "stream": False
    }
    try:
        response = requests.post(invoke_url, headers=headers, json=payload)
        response.raise_for_status()  # Raise HTTPError for bad responses (4xx or 5xx)
        return response.json()["choices"][0]["message"]["content"]
    except requests.exceptions.RequestException as e:
        return f"Error: Request failed - {e}"


if __name__ == "__main__":
    # Example usage
    prompt = "Please briefly explain the importance of fast AI inference."
    response = groq_completion(prompt)
    print(response)
